src/react/components/StyleForm.jsx

```diff
@@ -0,0 +1,348 @@
+            }
+        }
+    }
+
+    _getCurrentTab (tab) {
+        switch(tab) {
+            case 'point':
+                return (
+                    <PointTab
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        markers={ this.props.markers }
+                        styleSettings={ this.state }
+                    />
+                );
+            case 'line':
+                return (
+                    <LineTab
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        lineIcons={ lineIcons }
+                        styleSettings={ this.state }
+                    />
+                );
+            case 'area':
+                return (
+                    <AreaTab
+                        lineIcons={ lineIcons.lineDash }
+                        areaFills={ areaFills }
+                        formLayout={ formLayout }
+                        onChangeCallback={ (key, value) => this.updateState(key, value) }
+                        styleSettings={ this.state }
+                    />
+                );
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * @method _createPreview
+     * @description Compose Preview -component
+     * @returns {React.Component} Preview component with provided style
+     */
+    _createPreview () {
+        return (
+            <Row>
+                <Preview
+                    markers={ this.props.markers }
+                    fillPatterns={ areaFills }
+                    styleSettings={ this.state }
+                />
+            </Row>
+        );
+    }
+
+    /**
+     * @method _getColorPickers
+     * @description Compose fill and stroke color picker elements for style form
+     * 
+     * @returns {React.Component} Only stroke color picker or both stroke and fill 
+     */
+    _getColorPickers () {
+        return (
+            <Row>
+                <Form.Item name='stroke' label='Pisteen väri' { ...formLayout }>
+                    <ColorPicker
+                        onChange={ (event) => this.updateState('stroke.color', event.target.value) }
+                        defaultValue={ this.state.stroke.color } />
+                </Form.Item>
+
+                { this.state.format !== 'line' ?
+                    <Form.Item name='fill' label='Pisteen täyttöväri' { ...formLayout }>
+                        <ColorPicker
+                            onChange={ (event) => this.updateState('fill.color', event.target.value) }
+                            defaultValue={ this.state.fill.color }
+                        />
+                    </Form.Item>
+                    : false
+                }
+            </Row>
+        );
+    }
+
+    _getSizeControl () {
```

Comment:
Each form can init their own size controls so it shouldn't be done on the main form level